packages/connect/src/protocol-connect/handler-factory.ts

```diff
@@ -462,7 +462,7 @@ function createStreamHandler<I extends DescMessage, O extends DescMessage>(
       },
       transformSerializeEnvelope(serialization.getO(type.binary)),
       transformCatchFinally<EnvelopedMessage>((e) => {
-        context.abort();
+        context.abort(e);
```

Comment:
The connect handler has a separate code path for unary. For consistency, we have to call `context.abort(e)` in the catch block on line 261.